subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/support/KotlinCompiler.kt

```diff
@@ -375,18 +373,11 @@ fun CompilerConfiguration.addScriptDefinition(scriptDef: ScriptDefinition) {
 private
 fun Disposable.kotlinCoreEnvironmentFor(configuration: CompilerConfiguration): KotlinCoreEnvironment {
     org.jetbrains.kotlin.cli.common.environment.setIdeaIoUseFallback()
-    // Don't keep the Kotlin compiler environment alive as it might hold onto stale data.
-    // See https://youtrack.jetbrains.com/issue/KT-35394
```

Comment:
Wow, so fixing the fd leaks in kotlin would allow reusing the compiler context! The speed gain is a good incentive

Reply:
@bamboo, could you reproduce the file leaks with this removed?
In any case we should get back to Kotlin folks about it.
See https://youtrack.jetbrains.com/issue/KT-35394#focus=Comments-27-4118410.0-0

Reply:
I was able to reproduce the stale Kotlin definitions, see the `accessors are available after renaming precompiled script plugin from project dependency` test case changes in this PR.